main/src/com/google/refine/ProjectMetadata.java

```diff
@@ -123,23 +123,25 @@ public class ProjectMetadata {
 
     protected ProjectMetadata(LocalDateTime date) {
         _created = date;
+        lastSave = LocalDateTime.now();
         preparePreferenceStore(_preferenceStore);
     }
 
     public ProjectMetadata() {
         this(LocalDateTime.now());
         _modified = _created;
+        lastSave = LocalDateTime.now();
     }
 
     public ProjectMetadata(LocalDateTime created, LocalDateTime modified, String name) {
         this(created);
         _modified = modified;
         _name = name;
     }
-    
+
     @JsonIgnore
     public boolean isDirty() {
-        return written == null || _modified.isAfter(written);
+        return _modified.isAfter(lastSave);
```

Comment:
The case where `_modified` (previously, `written`) was `null` seemed to be an indication that the project medatada was never saved, so I am not sure it is safe to initialize it with `LocalDateTime.now()` in the constructor. If I create a brand new `ProjectMetadata` object, then I expect that it `isDirty()` because it has never been saved yet, right? Also, it is not clear to me that `_modified` is always initialized (in one of the constructors it is not) so I suspect there could be a Null Pointer Exception in some cases.